src/Polyfill/EmptyEscapeParser.php

```diff
@@ -150,7 +150,9 @@ private static function filterDocument($document)
     private static function extractRecord(): array
     {
         $record = [];
-        self::$line = self::$document->fgets();
+        if (false === (self::$line = self::$document->fgets())) {
```

Comment:
I would make the conditions more explicit ... 
```
if (false === self::$line) {
    return [null];
}
```

and that should be enough no ?

Reply:
I would agree but that seems to confuse PHPStan:

```
 ------ ------------------------------------------------------------------ 
  Line   src/Polyfill/EmptyEscapeParser.php                                
 ------ ------------------------------------------------------------------ 
  166    Strict comparison using !== between false and string will always  
         evaluate to true.                                                 
 ------ ------------------------------------------------------------------ 
```

See also: https://travis-ci.org/thephpleague/csv/jobs/594868542

It seems a bad idea to ignore the error completely but to ignore the error in just that file would require updating PHPStan which seems to open a can of worms (at least in my development environment).

This won't work with the current version of PHPStan. We need v0.11+:

```
	ignoreErrors:
		-
			message: '#Strict comparison using \!\=\= between false and string will always evaluate to true.#'
			path: src/Polyfill/EmptyEscapeParser.php
```

Reply:
we can not update because of the PHP7.0+ requirement but I have no issue ignoring this PHPstan notice, this is kind of a trade off. The day we drop PHP7.2- we will be able to upgrade/improve the ignore message IMHO.

Reply:
I think your patch is applied at the wrong lines 🤔 if you put the patch inside the while loop then PHPStan will not complained 👍 . You just need to patch the `$buffer` not so much the raw `self$line`.

```php
            $buffer = '';
            if (false !== self::$line) {
                $buffer = ltrim(self::$line, self::$trim_mask);
            }
```

Reply:
That's great - trade-off avoided!